plugin-k8s/src/k8s_probe.rs

```diff
@@ -52,57 +66,115 @@ impl alumet::pipeline::Source for K8SProbe {
+                        )
+                        .with_attr("pod", AttributeValue::String(metrics_gathered.name.clone()));
+                        measurements.push(p_usr);
+                    }
+                    if let Some(value_sys) = diff_sys {
+                        let p_sys: MeasurementPoint = MeasurementPoint::new(
+                            timestamp,
+                            self.metrics.time_used_system_mode,
+                            Resource::LocalMachine,
+                            consumer.clone(),
+                            value_sys as u64,
+                        )
+                        .with_attr("pod", AttributeValue::String(metrics_gathered.name.clone()));
+                        measurements.push(p_sys);
+                    }
+                },
+                Err(_) => {
+                    element_unreachable.push(metric_file.name.clone());
+                }
             };
-            let diff_usr = match counter_usr.update(metrics.time_used_user_mode) {
-                CounterDiffUpdate::FirstTime => None,
-                CounterDiffUpdate::Difference(diff) => Some(diff),
-                CounterDiffUpdate::CorrectedDifference(diff) => Some(diff),
-            };
-            let diff_sys = match counter_sys.update(metrics.time_used_system_mode) {
-                CounterDiffUpdate::FirstTime => None,
-                CounterDiffUpdate::Difference(diff) => Some(diff),
-                CounterDiffUpdate::CorrectedDifference(diff) => Some(diff),
-            };
-            let consumer = ResourceConsumer::ControlGroup {
-                path: (metric_file.path.to_string_lossy().to_string().into()),
-            };
-            if let Some(value_tot) = diff_tot {
-                let p_tot: MeasurementPoint = MeasurementPoint::new(
-                    timestamp,
-                    self.metrics.time_used_tot,
-                    Resource::LocalMachine,
-                    consumer.clone(),
-                    value_tot as u64,
-                )
-                .with_attr("pod", AttributeValue::String(metrics.name.clone()));
-                measurements.push(p_tot);
+        }
+        //Clean vector of all unreachable elements
+        for to_delete in  element_unreachable.iter(){
+            if let Some(index) = self.metric_and_counter.iter().position(|(metric_file, _, _, _)| metric_file.name == *to_delete) {
+                self.metric_and_counter.swap_remove(index);
             }
-            if let Some(value_usr) = diff_usr {
-                let p_usr: MeasurementPoint = MeasurementPoint::new(
-                    timestamp,
-                    self.metrics.time_used_user_mode,
-                    Resource::LocalMachine,
-                    consumer.clone(),
-                    value_usr as u64,
-                )
-                .with_attr("pod", AttributeValue::String(metrics.name.clone()));
-                measurements.push(p_usr);
+        }
```

Comment:
What if the failure is temporary?
How about testing the metrics when the plugin starts, like the nvidia plugin?